de.metas.business/src/main/java/de/metas/tax/api/impl/TaxDAO.java

```diff
@@ -124,4 +127,16 @@ public I_C_TaxCategory retrieveNoTaxCategoryFound(@CacheCtx final Properties ctx
 				.create()
 				.firstOnlyNotNull(I_C_TaxCategory.class);
 	}
+
+	@Override
+	public int findTaxCategoryId(IQueryFilter<I_C_TaxCategory> filter)
```

Comment:
introduce and use some query POJO class.
e.g. TaxCategoryQuery{ Boolean default; Boolean without; .... }.

as much as possible, keep the I_C_TaxCategory behind the DAO/Repo.